app/models/notify_user/scheduler.rb

```diff
@@ -14,11 +14,14 @@ def schedule
         return if aggregator.has_pending_deliveries?
 
         # Create the delivery:
-        delivery = Delivery.create!({
+        delivery_generator = DeliveryGenerator.for(channel)
+        options = {
           notification: notification,
           deliver_in: delay_time(aggregator),
           channel: channel.to_s
-        })
+        }
+
+        delivery_generator.generate(notification, options)
```

Comment:
I might inline the factory method here personally: `DeliveryGenerator.for(channel).generate(notification, options)`